src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -91,7 +91,7 @@ private static boolean isValidPath(Path filePath) {
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
+    public void save(AddressBook addressBook) throws StorageOperationException, FileNotFoundException {
```

Comment:
We were expecting you to do an explicit check if the file exists and if it does not to throw an exception. 
The exception should be thrown from StorageFile::save() method.
The best place to catch and terminate is Main::executeCommand() (this method already has a catch block to terminate upon exception.